### PR TITLE
add vaal encode & decoder

### PR DESCRIPTION
**Pull request by remo48**

### CODE REVIEW

**Review by AronDiSc — COMMENTED** [review]
Nice job. The KL divergent loss for the VAE bottleneck is missing though. We can do this later however.

From [VAAL](https://github.com/sinhasam/vaal/blob/master/solver.py):
`    
def vae_loss(self, x, recon, mu, logvar, beta):
        MSE = self.mse_loss(recon, x)
        KLD = -0.5 * torch.sum(1 + logvar - mu.pow(2) - logvar.exp())
        KLD = KLD * beta
        return MSE + KLD
`
